internal/toproto6/getresourceidentityschemas_test.go

```diff
@@ -0,0 +1,367 @@
+					"test_resource_1": {
+						IdentityAttributes: []*tfprotov6.ResourceIdentitySchemaAttribute{
+							{
+								RequiredForImport: true,
+								Name:              "test_attribute",
+								Type:              tftypes.Bool,
+							},
+						},
+					},
+					"test_resource_2": {
+						IdentityAttributes: []*tfprotov6.ResourceIdentitySchemaAttribute{
+							{
+								RequiredForImport: true,
+								Name:              "test_attribute",
+								Type:              tftypes.Bool,
+							},
+						},
+					},
+				},
+			},
+		},
+		"resource-identity-attribute-optionalforimport": {
+			input: &fwserver.GetResourceIdentitySchemasResponse{
+				IdentitySchemas: map[string]fwschema.Schema{
+					"test_resource": identityschema.Schema{
+						Attributes: map[string]identityschema.Attribute{
+							"test_attribute": identityschema.BoolAttribute{
+								OptionalForImport: true,
+							},
+						},
+					},
+				},
+			},
+			expected: &tfprotov6.GetResourceIdentitySchemasResponse{
+				IdentitySchemas: map[string]*tfprotov6.ResourceIdentitySchema{
+					"test_resource": {
+						IdentityAttributes: []*tfprotov6.ResourceIdentitySchemaAttribute{
+							{
+								Name:              "test_attribute",
+								OptionalForImport: true,
+								Type:              tftypes.Bool,
+							},
+						},
+					},
+				},
+			},
+		},
+		"resource-identity-attribute-requiredforimport": {
+			input: &fwserver.GetResourceIdentitySchemasResponse{
+				IdentitySchemas: map[string]fwschema.Schema{
+					"test_resource": identityschema.Schema{
+						Attributes: map[string]identityschema.Attribute{
+							"test_attribute": identityschema.BoolAttribute{
+								RequiredForImport: true,
+							},
+						},
+					},
+				},
+			},
+			expected: &tfprotov6.GetResourceIdentitySchemasResponse{
+				IdentitySchemas: map[string]*tfprotov6.ResourceIdentitySchema{
+					"test_resource": {
+						IdentityAttributes: []*tfprotov6.ResourceIdentitySchemaAttribute{
+							{
+								Name:              "test_attribute",
+								Type:              tftypes.Bool,
+								RequiredForImport: true,
+							},
+						},
+					},
+				},
+			},
+		},
+		"resource-identity-attribute-type-bool": {
+			input: &fwserver.GetResourceIdentitySchemasResponse{
+				IdentitySchemas: map[string]fwschema.Schema{
+					"test_resource": identityschema.Schema{
+						Attributes: map[string]identityschema.Attribute{
+							"test_attribute": identityschema.BoolAttribute{
+								RequiredForImport: true,
+							},
+						},
+					},
+				},
+			},
+			expected: &tfprotov6.GetResourceIdentitySchemasResponse{
+				IdentitySchemas: map[string]*tfprotov6.ResourceIdentitySchema{
+					"test_resource": {
+						IdentityAttributes: []*tfprotov6.ResourceIdentitySchemaAttribute{
+							{
+								Name:              "test_attribute",
+								RequiredForImport: true,
+								Type:              tftypes.Bool,
+							},
+						},
+					},
+				},
+			},
```

Comment:
This test looks identical to the one above – or am I lost in the code?
![image](https://github.com/user-attachments/assets/27b1e9db-8e0c-49af-ac0f-fc6c3b6d3c38)

Reply:
Seems like it's intentional though 🤔

Reply:
You're correct they are definitely the same, and likely not both needed. The intention with these tests is to just check the mapping, but here it's impossible to define an attribute without the type, so they end up the same.

(honestly there is a ton of duplicated tests in FW, we'll probably clean this up in a year or two once they start annoying us lol)